language-plutus-core/src/Language/PlutusCore/Renamer.hs

```diff
@@ -40,102 +37,84 @@ instance Monoid (TypeState a) where
 type TypeM a = StateT (TypeState a) (Either (RenameError a))
 
 -- | Annotate a PLC program, so that all names are annotated with their types/kinds.
-annotateProgramQ :: (MonadError (Error a) m, MonadQuote m) => Program TyName Name a -> m (Program TyNameWithKind NameWithType a)
-annotateProgramQ (Program a v t) = Program a v <$> annotateTermQ t
+annotateProgram :: (MonadError (Error a) m) => Program TyName Name a -> m (Program TyNameWithKind NameWithType a)
+annotateProgram (Program a v t) = Program a v <$> annotateTerm t
 
 -- | Annotate a PLC term, so that all names are annotated with their types/kinds.
-annotateTermQ :: forall m a . (MonadError (Error a) m, MonadQuote m) => Term TyName Name a -> m (Term TyNameWithKind NameWithType a)
-annotateTermQ t = convertErrors asError $ do
-    (ts, t') <- liftEither $ annotateTermST t
-    updateMaxU ts
+annotateTerm :: (MonadError (Error a) m) => Term TyName Name a -> m (Term TyNameWithKind NameWithType a)
+annotateTerm t = do
+    (t', _) <- liftEither $ convertError $ runStateT (annotateT t) mempty
```

Comment:
This could be `fmap fst $ liftEither $ convertError $ runStateT (annotateT t) mempty`.